packages/flutter_tools/lib/src/commands/create_utils.dart

```diff
@@ -0,0 +1,533 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:uuid/uuid.dart';
+
+import '../android/android.dart' as android_common;
+import '../android/android_workflow.dart';
+import '../android/gradle_utils.dart' as gradle;
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/utils.dart';
+import '../cache.dart';
+import '../convert.dart';
+import '../dart/pub.dart';
+import '../flutter_project_metadata.dart';
+import '../globals.dart' as globals;
+import '../project.dart';
+import '../runner/flutter_command.dart';
+import '../template.dart';
+
+/// Mixin contains methods that are shared with `flutter create` commands.
+mixin CreateCommandMixin on FlutterCommand {
+  /// Throw with exit code 2 if the output directory is invalid.
+  void validateOutoutDirectoryArg() {
+    if (argResults.rest.isEmpty) {
+      throwToolExit('No option specified for the output directory.\n$usage',
+          exitCode: 2);
+    }
+
+    if (argResults.rest.length > 1) {
+      String message = 'Multiple output directories specified.';
+      for (final String arg in argResults.rest) {
+        if (arg.startsWith('-')) {
+          message += '\nTry moving $arg to be immediately following $name';
+          break;
+        }
+      }
+      throwToolExit(message, exitCode: 2);
+    }
+  }
+
+  /// Gets the flutter root directory.
+  ///
+  /// Throw with exit code 2 if the flutter sdk installed is invalid.
+  String getFlutterRoot() {
+    if (Cache.flutterRoot == null) {
+      throwToolExit(
+          'Neither the --flutter-root command line flag nor the FLUTTER_ROOT environment '
```

Comment:
Remove the section about --flutter-root

```suggestion
          'Neither the --flutter-root command line flag nor the FLUTTER_ROOT environment '
```